avocado/utils/network/interfaces.py

```diff
@@ -128,6 +128,65 @@ def add_ipaddr(self, ipaddr, netmask):
         except Exception as ex:
             raise NWException("Failed to add address {}".format(ex))
 
+    @property
+    def vlans(self):
+        list_of_vlan_ids = {}
+        for i in self.host.interfaces:
+            details = i._get_interface_details()
+            if self.name == details.get("link"):
+                list_of_vlan_ids[details.get('linkinfo').get('info_data').get('id')] = details.get("ifname")
+        return list_of_vlan_ids
+
+    def add_vlan_tag(self, vlan_num, vlan_name=None):
+        """Configure 802.1Q VLAN tagging to the interface
+
+        This method will try to add a VLAN tag to this interface and return the
+        VLAN NetworkInterface object. If it fails, the method will raise a
+        NWException.
+
+        :param vlan_num: VLAN ID
+        :param vlan_name: option to name VLAN interface, by default it is named <interface_name>.<vlan_num>
+        """
+
+        vlan_name = vlan_name if vlan_name else "{}.{}".format(self.name, vlan_num)
+        cmd = "ip link add link {} name {} type vlan id {}".format(self.name, vlan_name,
+                                                                   str(vlan_num))
+        try:
+            run_command(cmd, self.host, sudo=True)
+        except Exception as ex:
+            raise NWException("Failed to add VLAN tag: {}".format(ex))
+
+    def remove_vlan_by_tag(self, vlan_num):
+        """Remove the VLAN of the interface by tag number
+
+        This method will try to remove the VLAN tag of this interface. If it fails,
+        the method will raise a NWException.
+
+        :param vlan_num: VLAN ID
+        """
+        if int(vlan_num) in self.vlans.keys():
+            vlan_name = self.vlans[int(vlan_num)]
```

Comment:
what if vlan_num is not in self.vlans.keys() ? vlan_name will not be defined.

Reply:
Wouldn't the method throw a `NWException` since the command will fail? 
```
 ip link delete None
Cannot find device "None"
```
or do you want to return gracefully with a `None` or a `False`?